src/main/groovy/edu/oregonstate/mist/slashcourses/resources/SlashCourseResource.groovy

```diff
@@ -74,16 +77,75 @@ class SlashCourseResource extends Resource {
      */
-//    @POST
-//    @Consumes(MediaType.TEXT_PLAIN)
-//    @Produces(MediaType.TEXT_PLAIN)
-//    public Response postMessage(String message, @Auth AuthenticatedUser authenticatedUser) {
-//        ResponseBuilder responseBuilder = ok(message)
-//        responseBuilder.build()
-//    }
+    @Path('{crn: \\d+}')
+    @DELETE
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response deleteByCRN (@PathParam('crn') Integer crn) {
+        slashCourseDAO.deleteByCRN(crn)
+        Response.ok().build()
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response postCourse (@Valid SlashCourse newCourse) {
+
+        Response returnResponse
+        try {
+
+            List<Integer> crnList = slashCourseDAO.getAllCRN()
+
+            if (newCourse.getInstructor() != null && !crnList.contains(newCourse.getCrn())) {
+                instructorDAO.postInstructor(newCourse.getInstructor().getLastName(),
+                                             newCourse.getInstructor().getFirstName())
+                slashCourseDAO.postCourse(newCourse.getCrn(),
+                                          newCourse.getCourseNum(),
+                                          newCourse.getCourseName(),
+                                          newCourse.getSlash(),
+                                          newCourse.getTerm(),
+                                          // retrieve the instructorId just created as instructorId in the new course object
+                                          instructorDAO.getLatestInstructorId().toInteger(),
+                                          newCourse.getDay(),
+                                          newCourse.getTime(),
+                                          newCourse.getLocation(),
+                                          newCourse.getType(),
+                                          newCourse.getInstructor())
+
+                URI createdURI = URI.create("/" + instructorDAO.getLatestInstructorId())
+                returnResponse = Response.created(createdURI).build()
+            } else {
+                slashCourseDAO.postCourse(newCourse.getCrn(),
+                                          newCourse.getCourseNum(),
+                                          newCourse.getCourseName(),
+                                          newCourse.getSlash(),
+                                          newCourse.getTerm(),
+                                          newCourse.getInstructorId(),
+                                          newCourse.getDay(),
+                                          newCourse.getTime(),
+                                          newCourse.getLocation(),
+                                          newCourse.getType(),
+                                          newCourse.getInstructor())
+                returnResponse = Response.ok().build()
+            }
```

Comment:
What type of validation does the POST method do? For example, what if I submit an empty instructor name or an invalid time, day, location, etc?